src/provider/metamask/MetaMask.ts

```diff
@@ -531,7 +533,7 @@ export class MetaMask {
   }
 
   async syncBlockchain(): Promise<MetaMaskSyncResult> {
-    const chainIdHex = (await this.getAPI()?.request({ method: "eth_chainId" })) as string;
+    const chainIdHex = (await (await this.getAPI())?.request({ method: "eth_chainId" })) as string;
```

Comment:
not sure if good practice to stack await maybe can seperate the api out?

Reply:
okay let me refactor it